src/planner/binder/tableref/bind_showref.cpp

```diff
@@ -151,6 +155,44 @@ unique_ptr<BoundTableRef> Binder::BindShowTable(ShowRef &ref) {
 		sql = PragmaShowDatabases();
 	} else if (lname == "\"tables\"") {
 		sql = PragmaShowTables();
+	} else if (ref.show_type == ShowType::SHOW_FROM) {
+		auto catalog_name = ref.catalog_name;
+		auto schema_name = ref.schema_name;
+
+		// First, try to resolve as a schema name
+		auto schema_entry = Catalog::GetSchema(context, catalog_name, schema_name, OnEntryNotFound::RETURN_NULL);
```

Comment:
Can we use `Binder::BindSchemaOrCatalog` here to figure out if something is a schema or a catalog? That also deals with detecting ambiguous references.

Reply:
Sure, can do. I was trying to see what `USE` did in this case, which uses a slightly different logic. Is it worth consolidating this with the Binder's approach? Or maybe out of scope for this case.


https://github.com/duckdb/duckdb/blob/59382ca326bd84009ee982657c17cef91054dd09/src/catalog/catalog_search_path.cpp#L162-L182

https://github.com/duckdb/duckdb/blob/59382ca326bd84009ee982657c17cef91054dd09/src/parser/transform/statement/transform_use.cpp#L7-L21